prow/tide/tide.go

```diff
@@ -231,17 +243,38 @@ func (sc *statusController) setStatuses(all, pool []PullRequest) {
 
 func (sc *statusController) run() {
 	lastStart := time.Time{} // Zero value so that we don't wait the first loop.
-	// wait for a new pool
-	for range sc.newPoolPending {
+	poolPending := false
+RunLoop:
+	for {
+		// wait for a new pool
+		if poolPending {
+			poolPending = false
+		} else if !<-sc.newPoolPending {
+			// chan was closed
+			break
+		}
 		// wait for the min sync period time to elapse if needed.
-		time.Sleep(time.Until(lastStart.Add(sc.ca.Config().Tide.StatusUpdatePeriod)))
-		lastStart = time.Now()
-
-		sc.Lock()
-		pool := sc.poolPRs
-		sc.Unlock()
-		sc.sync(pool)
+		wait := time.After(time.Until(lastStart.Add(sc.ca.Config().Tide.StatusUpdatePeriod)))
+	WaitLoop:
+		for {
+			select {
+			case <-wait:
+				lastStart = time.Now()
+				sc.Lock()
+				pool := sc.poolPRs
+				sc.Unlock()
+				sc.sync(pool)
+				break WaitLoop
+			case more := <-sc.newPoolPending:
+				if more {
+					poolPending = true
+				} else {
+					break RunLoop
+				}
+			}
+		}
 	}
+	sc.shutDown <- true
```

Comment:
generally you'd close the channel here after sending

Reply:
Yeah, that should be closed. I can actually close the channel **instead of** sending because a read from a closed channel returns immediately. Fixed.